src/Microsoft.AspNetCore.Mvc.Formatters.Json/MvcJsonOptionsExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Formatters.Json;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class MvcJsonOptionsExtensions
+    {
+        /// <summary>
+        /// Configures the casing behavior of JSON serialization to use camel case for property names, 
+        /// and optionally for dynamic types and dictionary keys.
+        /// </summary>
+        /// <remarks>
+        /// This method modifies <see cref="JsonSerializerSettings.ContractResolver"/>.
+        /// </remarks>
+        /// <param name="options"><see cref="MvcJsonOptions"/></param>
+        /// <param name="processDictionaryKeys">If true will camel case dictionary keys and properties of dynamic objects.</param>
+        /// <returns><see cref="MvcJsonOptions"/> with camel case settings.</returns>
+        public static MvcJsonOptions UseCamelCasing(this MvcJsonOptions options, bool processDictionaryKeys)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+            
+            if (options.SerializerSettings.ContractResolver is DefaultContractResolver resolver)
+            {
+                resolver.NamingStrategy = new CamelCaseNamingStrategy
+                {
+                    ProcessDictionaryKeys = processDictionaryKeys
+                };
+            }
+            else
+            {
+                var contractResolverName = options.SerializerSettings.ContractResolver.GetType().Name;
```

Comment:
I think MVC sets ContractResolver to a value on startup but someone could be set to null.

`options.SerializerSettings.ContractResolver?.GetType().Name;`

Probably should also add a unit test.

Reply:
Thank you for the feedback! That's a good point! What do you think about:

    if (options.SerializerSettings.ContractResolver == null)
    {
        throw new ArgumentNullException(nameof(options.SerializerSettings.ContractResolver));
    }

I'm thinking maybe it would be better to be specific about the null reference?

Reply:
I think an `InvalidOperationException` with a specific message would be better since resolver isn't an argument.